lib/resty/jwt.lua

```diff
@@ -452,28 +466,51 @@ end
 
 --@function validate_exp_nbf - validate expiry and not valid before
 --@param jwt_obj
-local function validate_exp_nbf(jwt_obj, leeway)
+local function validate_exp_nbf(jwt_obj, is_validation_loose, leeway)
+  if jwt_obj[str_const.reason] ~= nil then
+    return
+  end
+
   local exp = jwt_obj[str_const.payload][str_const.exp]
   local nbf = jwt_obj[str_const.payload][str_const.nbf]
 
-  if (exp ~= nil or nbf ~= nil ) and not jwt_obj[str_const.reason] then
-    leeway = leeway or 0
-    local now = ngx.now()
+  leeway = leeway or 0
+  local now = ngx.now()
+
+  if exp ~= nil and not is_validation_loose then
```

Comment:
@SkyLothar I'm actually not sure about the **raise 500** result.

Rationale is that this I don't see this as a server issue. The jwt is malformed and we requested a strict validation, thus, we should only reject the jwt.

IOW, I'm not sure to see the point of a 500 as it's there's nothing that can actually be done server side to "fix" the jwt.

Given this, we somehow end with the following behaviors (which I think is that the proposed code implements).

| present | loose validation | result |
|  :--: |  :---: | :--: |
| :white_check_mark: | :heavy_multiplication_x: | perform checking, fail the verification if malformed |
| :white_check_mark: | :white_check_mark:  | skip the checking |
| :heavy_multiplication_x:  | :heavy_multiplication_x:  | skip the checking |
| :heavy_multiplication_x:  | :white_check_mark:   | skip the checking |

It's also completely possible that I misunderstood your point or that there's an obvious flaw in my reasoning (That wouldn't be the first time :wink:)

Thoughts?